examples/llm/src/mosaic_gpt.py

```diff
@@ -205,13 +210,21 @@ class GPTMLP(nn.Module):
 
     def __init__(self, cfg: DictConfig, device: Optional[str] = None):
         super().__init__()
-        self.mlp_up = nn.Linear(cfg.d_model,
-                                cfg.mlp_ratio * cfg.d_model,
-                                device=device)
+        if te_installed and cfg.get('te_config.linear_only'):
```

Comment:
should the requisite warnings (e.g. if te_config provided but te not installed) but put here as well, if users are creating their own models and not using `main.py` ?